orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/manifest/DeployManifestStage.java

```diff
@@ -19,22 +19,38 @@
 
 import com.netflix.spinnaker.orca.clouddriver.tasks.MonitorKatoTask;
 import com.netflix.spinnaker.orca.clouddriver.tasks.artifacts.CleanupArtifactsTask;
+import com.netflix.spinnaker.orca.clouddriver.tasks.manifest.DeployManifestContext;
+import com.netflix.spinnaker.orca.clouddriver.tasks.manifest.DeployManifestContext.TrafficManagement;
 import com.netflix.spinnaker.orca.clouddriver.tasks.manifest.DeployManifestTask;
 import com.netflix.spinnaker.orca.clouddriver.tasks.manifest.ManifestForceCacheRefreshTask;
+import com.netflix.spinnaker.orca.clouddriver.tasks.manifest.ManifestHighlanderStrategy;
+import com.netflix.spinnaker.orca.clouddriver.tasks.manifest.ManifestNoneStrategy;
+import com.netflix.spinnaker.orca.clouddriver.tasks.manifest.ManifestRedBlackStrategy;
+import com.netflix.spinnaker.orca.clouddriver.tasks.manifest.ManifestStrategyType;
 import com.netflix.spinnaker.orca.clouddriver.tasks.manifest.PromoteManifestKatoOutputsTask;
 import com.netflix.spinnaker.orca.clouddriver.tasks.manifest.WaitForManifestStableTask;
 import com.netflix.spinnaker.orca.pipeline.StageDefinitionBuilder;
 import com.netflix.spinnaker.orca.pipeline.TaskNode;
+import com.netflix.spinnaker.orca.pipeline.graph.StageGraphBuilder;
 import com.netflix.spinnaker.orca.pipeline.model.Stage;
 import com.netflix.spinnaker.orca.pipeline.tasks.artifacts.BindProducedArtifactsTask;
+import javax.annotation.Nonnull;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Component;
 
 @Component
+@RequiredArgsConstructor
 public class DeployManifestStage implements StageDefinitionBuilder {
   public static final String PIPELINE_CONFIG_TYPE = "deployManifest";
 
+  private final ManifestHighlanderStrategy highlanderStrategy;
```

Comment:
Maybe another component that has a single function `addAfterStages` that takes the `Stage`, `StageGraphBuilder` and a `ManifestStrategyType` or something similar?  The idea being just to avoid needing each individual strategy autowired here (and anywhere else that wants to implement this).

Reply:
Good idea, added `ManifestStrategyStageAdder` with `addAfterStages` method!